scripts/deploy.sh

```diff
@@ -75,21 +114,83 @@ deploy() {
 
     address=$(echo -e "$output" | "$GREP" -oP '0x[0-9a-fA-F]+' | tail -n 1)
 
-    echo $address
+    echo "$address"
 }
 
 build
 
+if [[ $CORE_CLASS_HASH == "" ]]; then
+    core_class_hash=$(declare "$CORE_CONTRACT_SRC")
+else
+    core_class_hash=$CORE_CLASS_HASH
+fi
+
+if [[ $COMET_CLASS_HASH == "" ]]; then
+    comet_class_hash=$(declare "$COMET_CONTRACT_SRC")
+else
+    comet_class_hash=$COMET_CLASS_HASH
+fi
+
 if [[ $ERC20_CLASS_HASH == "" ]]; then
-    erc20_class_hash=$(declare $ERC20_CONTRACT_SRC)
+    erc20_class_hash=$(declare "$ERC20_CONTRACT_SRC")
 else
     erc20_class_hash=$ERC20_CLASS_HASH
 fi
 
 if [[ $ICS20_CLASS_HASH == "" ]]; then
-    ics20_class_hash=$(declare $ICS20_CONTRACT_SRC)
+    ics20_class_hash=$(declare "$ICS20_CONTRACT_SRC")
 else
     ics20_class_hash=$ICS20_CLASS_HASH
 fi
 
-deploy $ics20_class_hash $erc20_class_hash
+echo "Class hashes:"
+echo "  CORE: $core_class_hash"
+echo "  COMET: $comet_class_hash"
+echo "  ICS20: $ics20_class_hash"
+echo "  ERC20: $erc20_class_hash"
+
+core_contract_address=$(deploy_core "$core_class_hash")
+comet_contract_address=$(deploy_comet "$comet_class_hash" "$core_contract_address")
+ics20_contract_address=$(deploy_ics20 "$ics20_class_hash" "$core_contract_address" "$erc20_class_hash")
+
+echo "Contract addresses:"
+echo "  CORE: $core_contract_address"
+echo "  COMET: $comet_contract_address"
+echo "  ICS20: $ics20_contract_address"
+
+
+register_client() {
+    CORE_ADDRESS=$1
+    COMET_ADDRESS=$2
+
+    # client type uses cairo short-string
+    COMET_CLIENT_TYPE=$(starkli to-cairo-string "07-tendermint")
+
+    CALLDATA="$COMET_CLIENT_TYPE $COMET_ADDRESS"
+
+    starkli invoke \
+        $STARKLI_ARGS \
+        "$CORE_ADDRESS" \
+        "register_client" \
+        $CALLDATA
+}
+
+register_ics20() {
+    CORE_ADDRESS=$1
+    ICS20_ADDRESS=$2
+
+    TRANSFER_PORT=$(starkli to-cairo-string "transfer")
+
+    # https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/#serialization_of_byte_arrays
+    # there is 0 previous bytes, and `transfer` has 8 bytes
+    CALLDATA="0x0 $TRANSFER_PORT 0x8 $ICS20_ADDRESS"
+
+    starkli invoke \
+        $STARKLI_ARGS \
+        "$CORE_ADDRESS" \
+        "bind_port_id" \
+        $CALLDATA
+}
+
+register_client "$core_contract_address" "$comet_contract_address"
```

Comment:
Would be nice to echo a message after each `register_client`, and `register_ics20`.

Reply:
95f5b93